setuptools_rust/setuptools_ext.py

```diff
@@ -219,6 +219,19 @@ def get_tag(self):
         dist.cmdclass["bdist_wheel"] = bdist_wheel_rust_extension
 
 
+def patch_disutils_build():
+    from distutils.command import build as _build
+
+    class build(_build.build):
+        def finalize_options(self):
+            super().finalize_options()
+            if self.distribution.has_ext_modules():
+                self.build_lib = self.build_platlib
+            else:
+                self.build_lib = self.build_purelib
```

Comment:
Doing this will override any user-provided setting for `--build-lib`. Might be possible to avoid that by checking if `--build-lib` set:

```suggestion
            build_lib_user_specified = self.build_lib is not None
            super().finalize_options()
            if not build_lib_user_specified:
                if self.distribution.has_ext_modules():
                    self.build_lib = self.build_platlib
                else:
                    self.build_lib = self.build_purelib
```